extern/osg-ffmpeg-videoplayer/videostate.cpp

```diff
@@ -403,24 +402,29 @@ class VideoThread : public OpenThreads::Thread
             // Save global pts to be stored in pFrame
             global_video_pkt_pts = packet->pts;
             // Decode video frame
-            if(avcodec_decode_video2(self->video_ctx, pFrame, &frameFinished, packet) < 0)
+            int ret = avcodec_send_packet(self->video_ctx, packet);
+            // EAGAIN is not expected
+            if (ret < 0)
                 throw std::runtime_error("Error decoding video frame");
 
-            double pts = 0;
-            if(packet->dts != AV_NOPTS_VALUE)
-                pts = static_cast<double>(packet->dts);
-            else if(pFrame->opaque && *(int64_t*)pFrame->opaque != AV_NOPTS_VALUE)
-                pts = static_cast<double>(*(int64_t*)pFrame->opaque);
-            pts *= av_q2d(self->video_ctx->time_base);
+            while (!ret)
+            {
+                ret = avcodec_receive_frame(self->video_ctx, pFrame);
+                if (!ret)
+                {
+                    double pts = 0;
+                    if(packet->dts != AV_NOPTS_VALUE)
+                        pts = static_cast<double>(packet->dts);
+                    else if(pFrame->opaque && *(int64_t*)pFrame->opaque != AV_NOPTS_VALUE)
+                        pts = static_cast<double>(*(int64_t*)pFrame->opaque);
```

Comment:
You should be able to use the frame's `best_effort_timestamp` field here, rather than trying to decide between the `dts` field and what was stored in the `opaque` pointer. That should also allow getting rid of the `our_get_buffer` hack that was needed for older ffmpeg versions.

Reply:
Using the best_effort_timestamp now. A quick test did not show any issues.